activesupport/lib/active_support/duration/iso8601_duration_parser.rb

```diff
@@ -0,0 +1,61 @@
+# Parses a string formatted according to ISO 8601 Duration into the hash .
+#
+# See http://en.wikipedia.org/wiki/ISO_8601#Durations
+# Parts of code are taken from ISO8601 gem by Arnau Siches (@arnau).
+# This parser isn't so strict and allows negative parts to be present in pattern.
+class ActiveSupport::Duration::ISO8601DurationParser
```

Comment:
Let's do it this way:

```ruby
module ActiveSupport
  module Duration
    class ISO8601Parser
    end
  end
end
```

I'd also like the parser to be renamed as `ISO8601Parser`. It's already scoped to the `Duration` module, so I don't think we need to repeat that.